src/audio/module_adapter/module/generic.c

```diff
@@ -96,13 +96,24 @@ int module_init(struct processing_module *mod, struct module_interface *interfac
 	}
 
 	/*check interface, there must be one and only one of processing procedure */
-	if (!interface->init || !interface->prepare ||
-	    !interface->reset || !interface->free ||
-	    (!!interface->process + !!interface->process_audio_stream +
-	     !!interface->process_raw_data != 1)) {
-		comp_err(dev, "module_init(): comp %d is missing mandatory interfaces",
-			 dev_comp_id(dev));
-		return -EIO;
+	if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_LL) {
+		if (!interface->init || !interface->prepare ||
+		    !interface->reset || !interface->free ||
+		    (!!interface->process + !!interface->process_audio_stream +
+		     !!interface->process_raw_data != 1)) {
+			comp_err(dev, "module_init(): LL comp %d is missing mandatory interfaces",
+				 dev_comp_id(dev));
+			return -EIO;
+		}
+	} else {
+		/* DP processing */
+		if (!interface->init || !interface->prepare ||
+		    !interface->reset || !interface->free ||
+		    !interface->process || !interface->is_ready_to_process) {
```

Comment:
Minor nit: This could be more readable if the common checks for "init", "prepare", "reset" and "free" would be done first on L98 and then additional checks for LL and DP specifically. 
Not blocking, nice-to-have comment.